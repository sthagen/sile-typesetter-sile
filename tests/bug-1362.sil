\begin[papersize=a5]{document}
\use[module=packages.retrograde,target=v0.15.0]
\neverindent
\nofolios
\language[main=en]
\set[parameter=shaper.spacestretchfactor,value=1.2]% This 1.2 is the default
% value in SILE 0.12. It should have been 1 (see #1371) and if it ever changed later,
% we'd like the hyphenations below to occur at the same place...
\font[family=Libertinus Serif, features="+liga"]% We pick a font that has
% ligatures, and where those have a smaller width than the non-ligatured text.

% This text should be line-broken at "consecu-tively".
% We'd expect ligatures to apply (in "difficult", "effects", "affairs" and
% "finition") and yet the text to be also correctly justified.
% The partial workaround for #1362 does it...
Lorem ipsum dolor difficult effects affairs of finition lorem ipsum dolor sit amet
consecutively sadipscing elitr sed diam nonumy eirmod tempor invidunt ut labore
et dolore magna aliquyam

% This text should be line-broken at diffi-cult.
% We'd expect the line to be correctly justified, but also the ffi ligature in
% the hyphenated word to be applied. The workaround for #1362 is NOT sufficient
% for that...
xxxxxxxxx xxxxxxxxxxxxx xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx difficult
difficult affairs effects of finition lorem ipsum dolor sit amet
consetetur sadipscing elitr sed diam nonumy eirmod tempor invidunt ut labore

\end{document}
