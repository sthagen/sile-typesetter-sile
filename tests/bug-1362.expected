Set paper size 	419.5275636	595.275597
Begin page
Mx 	20.9764
My 	36.7438
Set font 	Libertinus Serif;10;400;;normal;+liga;;LTR
Mx 	5.2800
Mx 	5.0400
Mx 	3.6400
Mx 	4.4700
Mx 	7.9000
T	45 a=5.2800 80 a=5.0400 83 a=3.7200 70 a=4.4700 78 a=7.9000	(Lorem)
Mx 	50.9529
T	74 81 84 86 78 w=25.0100	(ipsum)
Mx 	79.6094
T	69 80 77 80 83 w=21.5000	(dolor)
Mx 	104.7559
T	69 74 2400 68 86 77 85 w=31.4500	(difficult)
Mx 	139.8525
Mx 	4.4700
Mx 	5.8200
Mx 	4.5400
Mx 	4.2800
Mx 	3.1600
Mx 	3.9000
T	70 a=4.4700 2397 a=5.8200 70 a=4.4700 68 a=4.2800 85 a=3.1600 84 a=3.9000	(effects)
Mx 	169.6690
T	66 2397 66 74 83 84 w=25.2900	(affairs)
Mx 	198.6055
T	80 71 w=8.1400	(of)
Mx 	210.3920
T	2398 79 74 85 74 80 79 w=30.0600	(finition)
Mx 	244.0986
Mx 	2.6400
Mx 	5.0400
Mx 	3.6400
Mx 	4.4700
Mx 	7.9000
T	77 a=2.6400 80 a=5.0400 83 a=3.7200 70 a=4.4700 78 a=7.9000	(lorem)
Mx 	271.4351
T	74 81 84 86 78 w=25.0100	(ipsum)
Mx 	300.0916
T	69 80 77 80 83 w=21.5000	(dolor)
Mx 	325.2381
T	84 74 85 w=9.7700	(sit)
Mx 	338.6547
T	66 78 70 85 w=20.1000	(amet)
Mx 	362.4012
T	68 80 79 w=14.7400	(con)
Mx 	377.1412
Mx 	3.9000
Mx 	4.5400
Mx 	4.2800
T	84 a=3.9000 70 a=4.4700 68 a=4.2800	(sec)
T	86 w=5.3100	(u)
Mx 	395.1712
T	14 w=3.3800	(-)
Mx 	20.9764
My 	48.7438
Mx 	3.1600
Mx 	2.7100
Mx 	4.8900
Mx 	4.4700
Mx 	2.6400
Mx 	5.1500
T	85 a=3.1600 74 a=2.7100 87 a=4.9700 70 a=4.4700 77 a=2.6400 90 a=5.1500	(tively)
Mx 	46.9975
T	84 66 69 74 81 84 68 74 79 72 w=42.7400	(sadipscing)
Mx 	92.7387
T	70 77 74 85 83 w=16.7000	(elitr)
Mx 	112.4398
Mx 	3.9000
Mx 	4.5400
Mx 	5.0600
T	84 a=3.9000 70 a=4.4700 69 a=5.0600	(sed)
Mx 	128.9409
T	69 74 66 78 w=20.2400	(diam)
Mx 	152.1821
T	79 80 79 86 78 90 w=34.2400	(nonumy)
Mx 	189.4232
Mx 	4.4700
Mx 	2.7100
Mx 	3.7200
Mx 	7.9000
Mx 	5.1100
Mx 	5.0600
T	70 a=4.4700 74 a=2.7100 83 a=3.7200 78 a=7.9000 80 a=5.0400 69 a=5.0600	(eirmod)
Mx 	221.3944
Mx 	3.1600
Mx 	4.4700
Mx 	7.9000
Mx 	5.2600
Mx 	5.0400
Mx 	3.7200
T	85 a=3.1600 70 a=4.4700 78 a=7.9000 81 a=5.1900 80 a=5.0400 83 a=3.7200	(tempor)
Mx 	253.9455
T	74 79 87 74 69 86 79 85 w=34.7600	(invidunt)
Mx 	291.7066
T	86 85 w=8.4700	(ut)
Mx 	303.1778
Mx 	2.6400
Mx 	4.5700
Mx 	5.0300
Mx 	5.0400
Mx 	3.6400
Mx 	4.4700
T	77 a=2.6400 66 a=4.5700 67 a=4.9300 80 a=5.0400 83 a=3.7200 70 a=4.4700	(labore)
Mx 	331.5689
T	70 85 w=7.6300	(et)
Mx 	342.2000
Mx 	5.0600
Mx 	5.0400
Mx 	2.6400
Mx 	5.0400
Mx 	3.6400
Mx 	4.4700
T	69 a=5.0600 80 a=5.0400 77 a=2.6400 80 a=5.0400 83 a=3.7200 70 a=4.4700	(dolore)
Mx 	371.0912
T	78 66 72 79 66 w=27.4600	(magna)
Mx 	20.9764
My 	60.7438
T	66 77 74 82 86 90 66 78 w=37.8800	(aliquyam)
My 	72.7438
Mx 	5.0000
Mx 	5.0000
Mx 	5.0000
Mx 	5.0000
Mx 	5.0000
Mx 	5.0000
Mx 	5.0000
Mx 	5.0000
Mx 	4.9000
T	89 a=4.9000 89 a=4.9000 89 a=4.9000 89 a=4.9000 89 a=4.9000 89 a=4.9000 89 a=4.9000 89 a=4.9000 89 a=4.9000	(xxxxxxxxx)
Mx 	70.2180
Mx 	5.0000
Mx 	5.0000
Mx 	5.0000
Mx 	5.0000
Mx 	5.0000
Mx 	5.0000
Mx 	5.0000
Mx 	5.0000
Mx 	5.0000
Mx 	5.0000
Mx 	5.0000
Mx 	5.0000
Mx 	4.9000
T	89 a=4.9000 89 a=4.9000 89 a=4.9000 89 a=4.9000 89 a=4.9000 89 a=4.9000 89 a=4.9000 89 a=4.9000 89 a=4.9000 89 a=4.9000 89 a=4.9000 89 a=4.9000 89 a=4.9000	(xxxxxxxxxxxxx)
Mx 	139.4596
Mx 	5.0000
Mx 	5.0000
Mx 	5.0000
Mx 	5.0000
Mx 	5.0000
Mx 	5.0000
Mx 	5.0000
Mx 	5.0000
Mx 	5.0000
Mx 	5.0000
Mx 	5.0000
Mx 	5.0000
Mx 	5.0000
Mx 	5.0000
Mx 	5.0000
Mx 	5.0000
Mx 	5.0000
Mx 	5.0000
Mx 	5.0000
Mx 	5.0000
Mx 	5.0000
Mx 	5.0000
Mx 	5.0000
Mx 	5.0000
Mx 	5.0000
Mx 	5.0000
Mx 	5.0000
Mx 	5.0000
Mx 	5.0000
Mx 	5.0000
Mx 	5.0000
Mx 	5.0000
Mx 	5.0000
Mx 	5.0000
Mx 	5.0000
Mx 	5.0000
Mx 	5.0000
Mx 	5.0000
Mx 	5.0000
Mx 	5.0000
Mx 	5.0000
Mx 	5.0000
Mx 	5.0000
Mx 	5.0000
Mx 	5.0000
Mx 	5.0000
Mx 	4.9000
T	89 a=4.9000 89 a=4.9000 89 a=4.9000 89 a=4.9000 89 a=4.9000 89 a=4.9000 89 a=4.9000 89 a=4.9000 89 a=4.9000 89 a=4.9000 89 a=4.9000 89 a=4.9000 89 a=4.9000 89 a=4.9000 89 a=4.9000 89 a=4.9000 89 a=4.9000 89 a=4.9000 89 a=4.9000 89 a=4.9000 89 a=4.9000 89 a=4.9000 89 a=4.9000 89 a=4.9000 89 a=4.9000 89 a=4.9000 89 a=4.9000 89 a=4.9000 89 a=4.9000 89 a=4.9000 89 a=4.9000 89 a=4.9000 89 a=4.9000 89 a=4.9000 89 a=4.9000 89 a=4.9000 89 a=4.9000 89 a=4.9000 89 a=4.9000 89 a=4.9000 89 a=4.9000 89 a=4.9000 89 a=4.9000 89 a=4.9000 89 a=4.9000 89 a=4.9000 89 a=4.9000	(xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx)
Mx 	378.7012
T	69 74 71 w=10.8700	(dif)
Mx 	389.5712
T	2398 w=5.6000	(fi)
Mx 	395.1712
T	14 w=3.3800	(-)
Mx 	20.9764
My 	84.7438
T	68 86 77 85 w=15.3900	(cult)
Mx 	39.3383
T	69 74 2400 68 86 77 85 w=31.4500	(difficult)
Mx 	73.7602
T	66 2397 66 74 83 84 w=25.2900	(affairs)
Mx 	102.0221
Mx 	4.4700
Mx 	5.8200
Mx 	4.5400
Mx 	4.2800
Mx 	3.1600
Mx 	3.9000
T	70 a=4.4700 2397 a=5.8200 70 a=4.4700 68 a=4.2800 85 a=3.1600 84 a=3.9000	(effects)
Mx 	131.1640
T	80 71 w=8.1400	(of)
Mx 	142.2759
T	2398 79 74 85 74 80 79 w=30.0600	(finition)
Mx 	175.3078
Mx 	2.6400
Mx 	5.0400
Mx 	3.6400
Mx 	4.4700
Mx 	7.9000
T	77 a=2.6400 80 a=5.0400 83 a=3.7200 70 a=4.4700 78 a=7.9000	(lorem)
Mx 	201.9697
T	74 81 84 86 78 w=25.0100	(ipsum)
Mx 	229.9516
T	69 80 77 80 83 w=21.5000	(dolor)
Mx 	254.4236
T	84 74 85 w=9.7700	(sit)
Mx 	267.1655
T	66 78 70 85 w=20.1000	(amet)
Mx 	290.2374
T	68 80 79 84 70 85 70 85 86 83 w=42.9300	(consetetur)
Mx 	336.1393
T	84 66 69 74 81 84 68 74 79 72 w=42.7400	(sadipscing)
Mx 	381.8512
T	70 77 74 85 83 w=16.7000	(elitr)
Mx 	20.9764
My 	96.7438
Mx 	3.9000
Mx 	4.5400
Mx 	5.0600
T	84 a=3.9000 70 a=4.4700 69 a=5.0600	(sed)
Mx 	37.5247
T	69 74 66 78 w=20.2400	(diam)
Mx 	60.8130
T	79 80 79 86 78 90 w=34.2400	(nonumy)
Mx 	98.1014
Mx 	4.4700
Mx 	2.7100
Mx 	3.7200
Mx 	7.9000
Mx 	5.1100
Mx 	5.0600
T	70 a=4.4700 74 a=2.7100 83 a=3.7200 78 a=7.9000 80 a=5.0400 69 a=5.0600	(eirmod)
Mx 	130.1197
Mx 	3.1600
Mx 	4.4700
Mx 	7.9000
Mx 	5.2600
Mx 	5.0400
Mx 	3.7200
T	85 a=3.1600 70 a=4.4700 78 a=7.9000 81 a=5.1900 80 a=5.0400 83 a=3.7200	(tempor)
Mx 	162.7181
T	74 79 87 74 69 86 79 85 w=34.7600	(invidunt)
Mx 	200.5264
T	86 85 w=8.4700	(ut)
Mx 	212.0447
Mx 	2.6400
Mx 	4.5700
Mx 	5.0300
Mx 	5.0400
Mx 	3.6400
Mx 	4.4700
T	77 a=2.6400 66 a=4.5700 67 a=4.9300 80 a=5.0400 83 a=3.7200 70 a=4.4700	(labore)
End page
Finish
